src/ClientRuntime/Microsoft.Rest.ClientRuntime.E2E.Tests/ScenarioTests/E2ETestBase.cs

```diff
@@ -0,0 +1,793 @@
+                    var dataDiskOut = vmOut.StorageProfile.DataDisks.FirstOrDefault(
+                            d => dataDisk.Lun == d.Lun);
+
+                    Assert.NotNull(dataDiskOut);
+                    Assert.Equal(dataDiskOut.DiskSizeGB, dataDisk.DiskSizeGB);
+                    Assert.Equal(dataDiskOut.CreateOption, dataDisk.CreateOption);
+                    if (dataDisk.Caching != null)
+                    {
+                        Assert.Equal(dataDiskOut.Caching, dataDisk.Caching);
+                    }
+
+                    if (dataDisk.Name != null)
+                    {
+                        Assert.Equal(dataDiskOut.Name, dataDisk.Name);
+                    }
+
+                    // Disabling resharper null-ref check as it doesn't seem to understand the not-null assert above.
+                    // ReSharper disable PossibleNullReferenceException
+
+
+                    Assert.NotNull(dataDiskOut.Vhd);
+                    Assert.NotNull(dataDiskOut.Vhd.Uri);
+                    if (dataDisk.Image != null && dataDisk.Image.Uri != null)
+                    {
+                        Assert.NotNull(dataDiskOut.Image);
+                        Assert.Equal(dataDisk.Image.Uri, dataDiskOut.Image.Uri);
+                    }
+                    // ReSharper enable PossibleNullReferenceException
+                }
+            }
+        }
+
+
+        protected PublicIPAddress CreatePublicIP(string rgName)
+        {
+            // Create publicIP
+            string publicIpName = "pip5913";
+            string domainNameLabel = "dn5913";
+
+            PublicIPAddress publicIp = null;
+
+            try
+            {
+                publicIp = NetworkClient.PublicIPAddresses.Get(rgName, publicIpName);
+            }
+            catch { }
+
+            if(publicIp == null)
+            {
+                publicIp = new PublicIPAddress()
+                {
+                    Location = DEFAULT_LOCATION,
+                    Tags = new Dictionary<string, string>()
+                    {
+                        {"key", "value"}
+                    },
+                    PublicIPAllocationMethod = IPAllocationMethod.Dynamic,
+                    DnsSettings = new PublicIPAddressDnsSettings()
+                    {
+                        DomainNameLabel = domainNameLabel
+                    }
+                };
+
+                var putPublicIpAddressResponse = NetworkClient.PublicIPAddresses.CreateOrUpdate(rgName, publicIpName, publicIp);
+                var getPublicIpAddressResponse = NetworkClient.PublicIPAddresses.Get(rgName, publicIpName);
+
+                publicIp = getPublicIpAddressResponse;
+            }
+            
+            return publicIp;
+        }
+
+        protected string CreateAvailabilitySet(string rgName, string asName, bool hasManagedDisks = false)
+        {
+            // Setup availability set
+            AvailabilitySet inputAvailabilitySet = null;
```

Comment:
@markcowl I modeled it after a simple test from compute. So I don't see a point in now removing it. Plus we don't know for sure at this stage if that is possible. So I am not going to experiment with this, as this is a really long running test and doing anything to experiment at this stage is not something I am planning on to do. If you feel strongly, we can create a issue and plan it.